### PR TITLE
Add int support for eq LLK

### LINKED ISSUE #20252: ttnn.eq unary produces incorrect results when input tensor is uint32 and tiled and when scalar is 0
**Describe the bug**
ttnn.eq unary produces incorrect results when input tensor is uint32 and tiled and when scalar is 0, which is causing low accuracy in distilbert model in pytorch2.0_ttnn project.

**To Reproduce**
Run unit test:
```
from loguru import logger
import random
import pytest
import torch
import ttnn
import traceback
from models.utility_functions import torch_random
from tests.ttnn.utils_for_testing import assert_with_pcc
aten = torch.ops.aten


def max_relative_error(expected, calculated):
    assert expected.shape == calculated.shape, "Tensors must have the same shape"
    if expected.dtype == torch.bool:
        expected = expected.to(torch.int)
    if calculated.dtype == torch.bool:
        calculated = calculated.to(torch.int)
    relative_error = torch.abs(expected - calculated) / (torch.abs(expected) + 1e-8)
    mre = torch.max(relative_error).item()
    inds = torch.where(relative_error == mre)
    return mre, inds


def run_repeat_tests(
    input_shape, 
    dtype, 
    dlayout,
    scalar,
    device,
):
    # All ones is input found in distilbert model
    # x = torch.ones(size=input_shape[0], dtype=torch.int32)
    # x = torch.randint(-100, 100, input_shape[0], dtype=torch.int32)
    x = torch.randint(-100, 100, input_shape[0], dtype=torch.int32)

    try:
        ref_value = aten.eq.Scalar(x, scalar).to(torch.int32)
        
        tt_x = ttnn.from_torch(x, dtype=dtype[0], layout=dlayout[0], device=device)

        tt_result = ttnn.eq(tt_x, scalar)
        tt_result = ttnn.to_torch(tt_result)
    except Exception as e:
        logger.warning(f"Test execution crashed: {e}")
        print(traceback.format_exc())
        raise e

    assert len(tt_result.shape) == len(ref_value.shape)
    assert tt_result.shape == ref_value.shape
    assert_with_pcc(ref_value, tt_result, 0.999)
    mre, inds = max_relative_error(ref_value, tt_result)
    assert mre < (0.0 + 1e-8), f"Maximum relative error is {mre}, {tt_result[inds]}"


test_sweep_args = [
    (
        [[1, 16]],
        [ttnn.uint32],
        [ttnn.TILE_LAYOUT],
        0,
    ),
]


@pytest.mark.parametrize(
    "input_shape, dtype, dlayout, scalar",
    (test_sweep_args),
)
def test_repeat(input_shape, dtype, dlayout, scalar, device):
    run_repeat_tests(input_shape, dtype, dlayout, scalar, device)

```

### PR DESCRIPTION
**Pull request by Aswinmcw**
### Ticket
#20252 

### Problem description
ttnn eq for scalar int32 inputs working incorrectly

### What's changed
Add int support for unary eq operation in LLK
For WH, negative comparison not working properly, giving incorrect result, hence alternate solution given
eq support provided for both float and int version

### Checklist
- [x] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
https://github.com/tenstorrent/tt-metal/actions/runs/14632995502
- [x] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
https://github.com/tenstorrent/tt-metal/actions/runs/14632997817
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models tests](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) CI passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h [hunk 1]
```diff
@@ -96,6 +96,12 @@ inline void calculate_comp_unary_int(int scalar) {
             v_endif;
             dst_reg[0] = val;
         }
+        // a[i] == scalar
+        else if constexpr (COMP_MODE == SfpuType::unary_eq) {
+            v_if(v == scalar) { val = 1; }
+            v_endif;
+            dst_reg[0] = val;
```
Comment (KalaivaniMCW) [inline tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h:103]: why is this being done twice ? once in the if-block and then again outside

Comment (Aswinmcw) [reply tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h:103]: second one unused, will remove it

**Comment by KalaivaniMCW** [comment]
Pls update the PR description

File: tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h (hunk 1 above)
Comment (KalaivaniMCW) [reply tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h:103]: the common code can be kept outside the if-block

**Review by VirdhatchaniKN — APPROVED** [review]

**Review by mcw-anasuya — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py [hunk 2]
```diff
@@ -589,22 +582,23 @@ def test_unary_eqz_edge_case(input_shapes, device):
...
+@pytest.mark.parametrize("ttnn_op", [ttnn.ne, ttnn.eq])
+def test_unary_comp_ops(input_shapes, scalar, ttnn_op, device):
     torch.manual_seed(213919)
 
     # Generate a uniform range of values across the valid int32 range
     num_elements = int(torch.prod(torch.tensor(input_shapes)).item())
     uniform_values = torch.linspace(-2147483647, 2147483647, num_elements, dtype=torch.int32)
 
-    corner_cases = torch.tensor([0, 1, -1, 2147483647, -2147483647], dtype=torch.int32)
+    corner_cases = torch.tensor([0, 1, -1, 2147483647, -2147483647, -100, -54, 13, 29], dtype=torch.int32)
```
Comment (VirdhatchaniKN) [inline tests/ttnn/unit_tests/operations/eltwise/test_unary.py:593]: Can you also update the pybind file?

File: tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_unary_comp.h [hunk 3]
```diff
@@ -33,6 +33,26 @@ inline void calculate_unary_ne(uint value) {
...
 }
 
+template <bool APPROXIMATION_MODE, int ITERATIONS = 8>
+inline void calculate_unary_eq(uint value) {
+    // SFPU microcode
+    Converter c_value;
+    c_value.u = value;
+    vFloat s = c_value.f;
+
+#pragma GCC unroll 0
```
Comment (patrickroberts) [inline tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_unary_comp.h:43]: Is there a reason we want to block unrolling?
> The values of 0 and 1 block any unrolling of the loop.

From https://gcc.gnu.org/onlinedocs/gcc/Loop-Specific-Pragmas.html

Comment (rtawfik01) [reply tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/ckernel_sfpu_unary_comp.h:43]: Only reason would be if it impacts code size to the point where it causes build failures

File: tests/ttnn/unit_tests/operations/eltwise/test_unary.py (hunk 2 above)
Comment (Aswinmcw) [reply tests/ttnn/unit_tests/operations/eltwise/test_unary.py:614]: Will be updated in @mouliraj-mcw PR

Comment (VirdhatchaniKN) [reply tests/ttnn/unit_tests/operations/eltwise/test_unary.py:614]: Can you mention that your PR supports for Tensor-scalar version of the op? Because https://github.com/tenstorrent/tt-metal/pull/21048 mentions its supported version. But on rebasing, based on which PR Goes to main first, the second PR should have the line removed. We do not want a wrong detail in the documentation.

**Review by dchenTT — APPROVED** [review]

**Review by umadevimcw — APPROVED** [review]

**Review by nvelickovicTT — APPROVED** [review]

**Review by VirdhatchaniKN — APPROVED** [review]

File: tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h [hunk 4]
```diff
@@ -132,7 +132,7 @@ inline void calculate_comp_unary_int(int scalar) {
             v_if(v >= 0) {
                 v_if(v != scalar) { val = 1; }
                 v_endif;
-            }
+            }  // negative comparison not working as expected in WH hence alternate implementation
```
Comment (KalaivaniMCW) [inline tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_comp.h:135]: @rdjogoTT @rtawfik01 Is this a known limitation for INT32 in Wormhole ?

**Review by KalaivaniMCW — APPROVED** [review]
